[Docs] Translate model analysis document to English (#956)
open-mmlab/mmengine: docs/en/common_usage/mode_analysis.md

> # Count the calculation of Model and Parameter

Reviewer: ```suggestion
# Calculate the FLOPs and Parameters of Model
```

1. Define Model

```python
import torch.nn.functional as F
import torchvision
from mmengine.model import BaseModel

class MMResNet50(BaseModel):
    def __init__(self):
        super().__init__()
        self.resnet = torchvision.models.resnet50()

    def forward(self, imgs, labels=None, mode='tensor'):
        x = self.resnet(imgs)
        if mode == 'loss':
            return {'loss': F.cross_entropy(x, labels)}
        elif mode == 'predict':
            return x, labels
        elif mode == 'tensor':
            return x

> ```

Reviewer: Need 2 blank indentations

Author: mdformat.................................................................Failed
- hook id: mdformat
- files were modified by this hook

if using 2 blank indentation, pre-commit errors. i find that some docs using 3 blank indentation. so i used

Reviewer: OK

> 2. Count the number of calculation and parameter

Reviewer: ```suggestion
2. Calculate the FLOPs and Parameters
```

```python
from mmengine.analysis import get_model_complexity_info

input_shape = (3, 224, 224)
model = MMResNet50()
analysis_results = get_model_complexity_info(model, input_shape)

> ```

Reviewer: Need 2 blank indentations

- Using table form to show

```python
print(analysis_results['out_table'])

> ```

Reviewer: Need 2 blank indentations

> - Using table form to show

Reviewer: ```suggestion
- Show in table form
```

> <details>
>     <summary>Click to expand</summary>

Reviewer: ```suggestion
  <details>
  <summary>Click to expand</summary>
```

|    layer4.1.bn2        |    1.024K            |    0.125M  |    0         |
|    layer4.1.conv3      |    1.049M            |    51.38M  |    0.1M      |
|    layer4.1.bn3        |    4.096K            |    0.502M  |    0         |
|   layer4.2             |   4.463M             |   0.219G   |   0.151M     |
|    layer4.2.conv1      |    1.049M            |    51.38M  |    25.088K   |
|    layer4.2.bn1        |    1.024K            |    0.125M  |    0         |
|    layer4.2.conv2      |    2.359M            |    0.116G  |    25.088K   |
|    layer4.2.bn2        |    1.024K            |    0.125M  |    0         |
|    layer4.2.conv3      |    1.049M            |    51.38M  |    0.1M      |
|    layer4.2.bn3        |    4.096K            |    0.502M  |    0         |
|  fc                    |  2.049M              |  2.048M    |  1K          |
|   fc.weight            |   (1000, 2048)       |            |              |
|   fc.bias              |   (1000,)            |            |              |
|  avgpool               |                      |  0.1M      |  0           |
+------------------------+----------------------+------------+--------------+

> ```

Reviewer: Need 2 blank indentations

</details>

- Using Model structure to show

```python
print(analysis_results['out_arch'])
```

<details>

>     <summary>点击展开</summary>

Reviewer: ```suggestion
    <summary>Click to expand</summary>
```

        512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False
        #params: 2.36M, #flops: 0.12G, #acts: 25.09K
        )
        (bn2): BatchNorm2d(
        512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True
        #params: 1.02K, #flops: 0.13M, #acts: 0
        )
        (conv3): Conv2d(
        512, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False
        #params: 1.05M, #flops: 51.38M, #acts: 0.1M
        )
        (bn3): BatchNorm2d(
        2048, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True
        #params: 4.1K, #flops: 0.5M, #acts: 0
        )
        (relu): ReLU(inplace=True)
    )
    )
    (avgpool): AdaptiveAvgPool2d(
    output_size=(1, 1)
    #params: 0, #flops: 0.1M, #acts: 0
    )
    (fc): Linear(
    in_features=2048, out_features=1000, bias=True
    #params: 2.05M, #flops: 2.05M, #acts: 1K
    )
)
)
```

</details>

- Total number of calculation

```python
print("Model Flops:{}".format(analysis_results['flops_str']))
# Model Flops:4.145G
```

- Total number of Parameters

```python
print("Model Parameters:{}".format(analysis_results['params_str']))
# Model Parameters:25.557M
```

> For the definition of model calculation and parameter and more usage, please refer[Model Complexity Analysis](../advanced_tutorials/model_analysis.md)

Reviewer: ```suggestion
For the definition of FLOPs and Parameters of model and more usage, please refer [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```

> - Total number of calculation

Reviewer: ```suggestion
- Show FLOPs as a string
```

> - Total number of Parameters

Reviewer: ```suggestion
- Show Parameters as a string
```

> For the definition of FLOPs and Parameters of model and more usage, please refer [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)

Reviewer: ```suggestion
For the definition of FLOPs and Parameters of model and more usage, please refer to [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```
